pkg/liquidity-mining/test/GaugeWorkingBalanceHelper.test.ts

```diff
@@ -0,0 +1,339 @@
+      expect(await workingBalanceHelper.getVotingEscrow()).to.equal(votingEscrow.address);
+    });
+
+    it('indicates where to read supply from', async () => {
+      expect(await workingBalanceHelper.readsTotalSupplyFromVE()).to.equal(readTotalSupplyFromVE);
+    });
+  }
+
+  function itComputesWorkingBalances() {
+    const TOKENLESS_PRODUCTION = 0.4;
+    const MAX_BALANCE_RATIO = 1 / TOKENLESS_PRODUCTION;
+    const LOCK_PERIOD = 365 * DAY;
+    const stakeAmount = fp(5);
+    const bptAmount = fp(10);
+
+    async function createLockForUser(
+      account: SignerWithAddress,
+      amount: BigNumberish,
+      lockDuration: BigNumberish
+    ): Promise<void> {
+      await bpt8020.mint(account, amount);
+      await bpt8020.approve(votingEscrow, amount, { from: account });
+      const now = await currentTimestamp();
+      await votingEscrow.connect(account).create_lock(amount, now.add(lockDuration));
+    }
+
+    async function depositIntoGauge(account: SignerWithAddress, stakeAmount: BigNumberish) {
+      await pool.connect(admin).mint(account.address, stakeAmount);
+      await pool.connect(account).approve(gauge.address, stakeAmount);
+      await gauge.connect(account)['deposit(uint256)'](stakeAmount);
+    }
+
+    function currentBalanceEqualsProjected() {
+      it('projected balance should equal current', async () => {
+        const [currentWorkingBalance, projectedWorkingBalance] = await workingBalanceHelper.getWorkingBalances(
+          gauge.address,
+          user.address
+        );
+
+        // Ensure we have equal balances (that are non-zero)
+        expect(projectedWorkingBalance).to.eq(currentWorkingBalance);
+        expect(projectedWorkingBalance).to.gt(0);
+      });
+    }
+
+    function currentRatioEqualsProjected() {
+      it('current and projected ratios should be 1', async () => {
+        const [current, projected] = await workingBalanceHelper.getWorkingBalanceToSupplyRatios(
+          gauge.address,
+          user.address
+        );
+
+        // Ensure we have equal ratios (that equal 1)
+        expect(projected).to.eq(current);
+        expect(projected).to.eq(FP_ONE);
+      });
+    }
+
+    function projectedRatioIncreases() {
+      it('current and projected ratios should be 1', async () => {
+        const [current, projected] = await workingBalanceHelper.getWorkingBalanceToSupplyRatios(
+          gauge.address,
+          user.address
+        );
+
+        expect(projected).to.gt(current);
+      });
+    }
+
+    function projectedRatioIsMaxedOut() {
+      it(`projected ratio should be greater than current by the maximum ratio (${MAX_BALANCE_RATIO})`, async () => {
+        const [currentWorkingBalance, projectedWorkingBalance] = await workingBalanceHelper.getWorkingBalances(
+          gauge.address,
+          user.address
+        );
+
+        expect(fromFp(projectedWorkingBalance) / fromFp(currentWorkingBalance)).to.eq(MAX_BALANCE_RATIO);
```

Comment:
`fromFp` returns a `Decimal`; can we use `/` just like that?

Shouldn't it be something like
```
fromFp(projectedWorkingBalance).div(fromFp(currentWorkingBalance))
```

Reply:
Maybe there's another way to do it, but div like that seems to truncate it, and the test fails.

Reply:
```     expect(fromFp(projectedWorkingBalance).div(fromFp(currentWorkingBalance)).toNumber()).to.eq(MAX_BALANCE_RATIO);
```

works for me. It's a bit more verbose, but I get no warnings in the IDE.